### PR TITLE
execution: fix index's duplicate error message

### LINKED ISSUE #22851: prefixed column clustered index's duplicate error message differ to mysql
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
set @@tidb_enable_clustered_index=1;
create table t(name varchar(255), b int, c int, primary key(name(2)));
insert into t(name, b) values("cha", 3);
insert into t(name, b) values("chb", 3);
```

### 2. What did you expect to see? (Required)

```
ERROR 1062 (23000): Duplicate entry 'ch' for key 'PRIMARY'
```

### 3. What did you see instead (Required)

```
ERROR 1062 (23000): Duplicate entry 'chb' for key 'PRIMARY'
```

### 4. What is your TiDB version? (Required)

```
Release Version: None
Edition: Community
Git Commit Hash: None
Git Branch: None
UTC Build Time: None
GoVersion: go1.15
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by b41sh**
### What problem does this PR solve?

Issue Number: close #22851 

Problem Summary: index duplicate error message is incompatible with MySQL

### What is changed and how it works?

What's Changed: check column length and delete strings of extra length

How it Works:

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- duplicate error message compatible with MySQL

### CODE REVIEW

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -431,6 +431,9 @@ func (c *twoPhaseCommitter) extractKeyExistsErrFromHandle(key kv.Key, value []by
 		if err != nil {
 			return c.genKeyExistsError(name, key.String(), err)
 		}
+		if len([]rune(str)) > col.Length {
+			str = string([]rune(str)[:col.Length])
```
Comment (lysu) [inline store/tikv/2pc.go:435]: `col.Length` maybe be `-1` when it's `col1` without `(2)` :smile:

Comment (b41sh) [reply store/tikv/2pc.go:435]: fixed, PTAL @lysu

**Pull request merged**